### PR TITLE
docs(init): leading-underscore keys are aliased, not kept bare (#467)

### LINKED ISSUE #467: Merged #443 follow-ups: partial-encryption still bypasses secret-lockout guard (#457) + init.md isidentifier docs stale (#460)
_🤖 Filed by the envdrift review-bot (via Jainal's account)._

Two review-bot findings from the #443 wave merged without being addressed. Both still reproduce on `main`.

---

## 1. Secret-lockout guard is backend-only — partial-encryption (`push`/`lock`) still bypasses it  ·  _medium · data-loss_

Originally raised on #457 (merged `6a60a63`). #457 added a filename-slug guard to **`DotenvxEncryptionBackend.encrypt`** (`src/envdrift/encryption/dotenvx.py`, the `[A-Za-z0-9._-]` check), and the merge also routed `DotenvxWrapper.encrypt` through `self._validate_filename`. **But `_validate_filename` only rejects the Windows `.env.local` pattern**, not space/non-ASCII slugs:

```python
# src/envdrift/integrations/dotenvx.py (main)
PROBLEMATIC_FILENAME_PATTERNS = [
    r"^\.env\.local$",          # <- the ONLY pattern
]
def _validate_filename(file_path):
    for pattern in PROBLEMATIC_FILENAME_PATTERNS:
        if re.match(pattern, file_path.name.lower(), re.IGNORECASE):
            raise DotenvxFilenameError(...)
```

`DotenvxWrapper.encrypt` has no `[A-Za-z0-9._-]` slug check of its own. The partial-encryption path uses the wrapper, not the guarded backend:

- `encrypt_secret_file` → `DotenvxWrapper().encrypt(secret_file)` (`src/envdrift/core/partial_encryption.py`)
- `_encrypt_secrets_only_file` (from `push_secrets_only`) → `dotenvx.encrypt(file)`, where `file` comes from `secrets_dir.glob(pattern)` — **arbitrary on-disk filenames**.

So a user with e.g. `my secret.env` or `café.env.secret` in their `secrets_dir` who runs `envdrift push` / `lock` (secrets-only) still hits the exact permanent-lockout this PR set out to fix: the value encrypts, dotenvx exits 0, the plaintext is destroyed, and the derived `DOTENV_PRIVATE_KEY_<bad slug>` is unusable so decrypt can never recover it. (The original finding empirically confirmed `roundtrip=0` for such names with dotenvx.) The existing regression tests cover the **backend** path (`test_encrypt_*` with `my secrets.env`/`café.env`) and café-in-*content*, but none drives the **partial-encryption** path with a space/unicode *filename*.

**Fix:** share the slug-validity guard so `DotenvxWrapper.encrypt` enforces it too (or have the partial-encryption callers route through the guarded backend), and add a regression test that drives `push_secrets_only`/`encrypt_secret_file` with a space/unicode-named secret file and asserts it refuses **with plaintext preserved**.

## 2. `docs/cli/init.md` `isidentifier()` claim contradicts the new leading-underscore aliasing  ·  _low · docs_

Originally raised on #460 (merged `ab127ad`). #460 made `init` sanitize a key that natively starts with `_` (e.g. `_PRIVATE` → `field__PRIVATE: str = Field(alias='_PRIVATE')`). But `docs/cli/init.md:178` still states:

> A key that passes `str.isidentifier()` (e.g. `CAFÉ`, `ΑΛΦΑ`) becomes a bare field with no alias.

A leading-underscore identifier passes `str.isidentifier()` yet is now aliased, so the claim is an over-generalization. Update the bullet (and add a leading-underscore example) so the docs match the shipped behavior (CLAUDE.md: keep `docs/` in sync with behavior changes).

### PR DESCRIPTION
**Pull request by jainal09**
## Part 2 of #467 — stale `isidentifier()` claim in `docs/cli/init.md` (low · docs)

#460 made `init` sanitize a key that natively starts with `_`:

```python
field__PRIVATE: str = Field(alias='_PRIVATE')   # _PRIVATE, verified live
```

because Pydantic rejects field names with a leading underscore ("Fields must not use names with
leading underscores"). But `docs/cli/init.md` still claimed:

> A key that passes `str.isidentifier()` (e.g. `CAFÉ`, `ΑΛΦΑ`) becomes a bare field with no alias.

A leading-underscore identifier *passes* `str.isidentifier()` yet is now aliased, so the claim was an
over-generalization that contradicted shipped behavior.

### Fix
- Qualify the "kept verbatim" bullet to **exclude** leading-underscore keys.
- Add a dedicated bullet documenting the aliasing, with the real `field__PRIVATE` example.

### Test (real, no mock)
A new `test_docs_accuracy` regression drives the **real** `_sanitize_identifier`
(`_PRIVATE` → `field__PRIVATE`, `CAFÉ` → `CAFÉ`) and pins the doc to that output, so the claim can't
drift back. It fails RED against the pre-fix doc. `make lint-docs` clean (59 files, 0 errors).

Closes #467 (paired with #468, which fixes finding #1 — the partial-encryption data-loss bypass).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR tightens the \"valid non-ASCII identifiers are kept verbatim\" bullet in `docs/cli/init.md` to exclude two additional aliasing cases that #460 introduced: keys with a leading underscore, and keys that pass `str.isidentifier()` but whose NFKC-normalized form differs from the raw key (NFD compositions, ligatures). Two regression tests are added to pin the doc to the real `_sanitize_identifier` / `_resolve_field_name` output.

- **`docs/cli/init.md`**: The \"kept verbatim\" bullet now correctly requires the key to be both leading-underscore-free and already NFKC-normalized; a new bullet documents the `_PRIVATE` → `field__PRIVATE` aliasing with a concrete code example.
- **`tests/unit/test_docs_accuracy.py`**: Two new tests drive the real sanitizer/resolver functions against known inputs (NFC/NFD `CAFÉ`, `ﬁle`, `_PRIVATE`) and assert the doc contains the updated prose, preventing silent doc drift.

<h3>Confidence Score: 5/5</h3>

Safe to merge — changes are confined to documentation prose and regression tests; no production logic is touched.

The doc update correctly narrows the kept-verbatim guarantee to match what _resolve_field_name actually does. The two new tests drive the real sanitizer/resolver against concrete inputs and their assertions match the implementation's return values, so tests would catch any future drift.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| docs/cli/init.md | Documentation accurately updated: 'kept verbatim' bullet now carves out leading-underscore and non-NFKC-normalized keys; new bullet with field__PRIVATE example matches real sanitizer output. |
| tests/unit/test_docs_accuracy.py | Two new regression tests exercise the real _sanitize_identifier and _resolve_field_name functions; assertions on NFC/NFD CAFÉ, the ﬁ ligature, and _PRIVATE all match the implementation's actual return values. |

</details>

<sub>Reviews (2): Last reviewed commit: ["docs(init): qualify the kept-verbatim bu..."](https://github.com/jainal09/envdrift/commit/822bddd55f88589189b2a5bfb70521b1311f47b7) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=36728312)</sub>

### CODE REVIEW

File: docs/cli/init.md [hunk 1]
```diff
@@ -175,7 +175,18 @@ The generated module is always valid, importable Python:
   ```
 
 - **Valid non-ASCII identifiers are kept verbatim.** A key that passes
-  `str.isidentifier()` (e.g. `CAFÉ`, `ΑΛΦΑ`) becomes a bare field with no alias.
+  `str.isidentifier()` and does not start with `_` (e.g. `CAFÉ`, `ΑΛΦΑ`) becomes
```
Comment (jainal09) [inline docs/cli/init.md]: 🤖 **envdrift review-bot** · low · doc-mismatch


The PR narrows the bullet to:

> A key that passes `str.isidentifier()` and does not start with `_` (e.g. `CAFÉ`, `ΑΛΦΑ`) becomes a bare field with no alias.

This is accurate for the NFC examples shown, but it is still falsified by a class of keys the very function under discussion handles explicitly. In `src/envdrift/cli_commands/init_cmd.py`, `_resolve_field_name` aliases any key whose NFKC fold differs from the raw key:

```python
alias = var_name if (sanitize or bumped or _nfkc(field_name) != var_name) else None
```

I ran the head code: NFD-form `CAFÉ` (`'CAFÉ'`) passes `isidentifier()`, does not start with `_`, is not a keyword or pydantic-reserved name — yet `_resolve_field_name` returns `('CAFÉ', 'CAFÉ')` (aliased, because Python NFKC-folds identifiers at compile time). Same for the ligature `ﬁle` → `('ﬁle', 'ﬁle')`. Unlike the keyword/reserved cases (covered by earlier bullets in the same list), no other bullet covers this case, so a reader following the case analysis lands on this bullet and gets the wrong answer. This is the same shape of over-generalization #467 Part 2 complained about, surviving in the edited sentence.

Suggested direction: qualify the bullet further, e.g. "...and is already NFKC-normalized (Python folds identifiers with NFKC at compile time; an NFD-form or ligature key is aliased to preserve the exact original name)", or fold a one-line note into the round-trip bullet. The `_resolve_field_name` docstring already describes this behavior and can be quoted.

**Comment by jainal09** [comment]
🤖 **envdrift review-bot** — automated re-review of commit a005ecd


The amended doc wording is much tighter than before, but one verified low-severity over-generalization remains in the "kept verbatim" bullet of `docs/cli/init.md`.

**Findings (1):**
- [low] Amended 'kept verbatim' bullet is still over-general: NFKC-non-normalized identifiers (NFD CAFÉ, ligatures) get an alias despite passing isidentifier() and not starting with _ — `docs/cli/init.md:178` (inline)

File: docs/cli/init.md (hunk 1 above)
Comment (jainal09) [reply docs/cli/init.md]: Good catch — confirmed live against the real resolver: NFC `CAFÉ` → bare, NFD `CAFÉ` → `('CAFÉ', 'CAFÉ')` (aliased), ligature `ﬁle` → aliased, exactly per `_resolve_field_name`'s NFKC-fold condition.

Fixed in 822bddd: the bullet now requires the key to be **already NFKC-normalized** and explains why (Python NFKC-folds identifiers at compile time; an NFD-form or ligature key is aliased to preserve the exact original name). Added `test_init_md_qualifies_nfkc_aliasing`, which pins both the real resolver behavior (NFC bare, NFD/ligature aliased) and the doc wording — it fails RED against the previous text, so the case analysis can't regress to the over-general claim again.
